user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -1131,8 +1125,54 @@ define([
       var $target = $(event.currentTarget).parent().parent();
```

Comment:
I think adding comment explaining the if block here would help. What each case handles.